### PR TITLE
Infer bug_tracking_url and code_view_url from vcs_url

### LINKED ISSUE #4943: Infer `bug_tracking_url` and `code_view_url` from `vcs_url` for major code hosting platforms
### Description

> Please leave a brief description of the bug or feature request:

Could be done for:

- Github
- Gitlab
- Codeberg
- Bitbucket

Could be a package post-parsing step like we do license detections post file parsing.

### PR DESCRIPTION
**Pull request by Kaushik-Kumar-CEG**
Fixes #4943

This PR adds a post-parsing step to automatically infer `bug_tracking_url` and `code_view_url` from a package's `vcs_url` if they aren't already provided in the manifest. 

It handles standard HTTPS links, as well as `git@` SSH strings and `git+` prefixes, stripping trailing `.git` suffixes so the final URLs resolve properly in a browser.

Added tests covering all 4 requested platforms (GitHub, GitLab, Codeberg, and Bitbucket) alongside the URL edge cases.

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁
* [ ] Updated documentation pages (if applicable)
* [ ] Updated CHANGELOG.rst (if applicable)

### CODE REVIEW

File: src/packagedcode/models.py [hunk 1]
```diff
@@ -768,6 +768,8 @@ def from_data(cls, package_data, package_only=False):
         else:
             package_data.normalize_extracted_license_statement()
 
+        package_data.infer_vcs_urls()
```
Comment (AyanSinhaMahapatra) [inline src/packagedcode/models.py:771]: Yes, this is exactly where we want to process this.

File: src/packagedcode/utils.py [hunk 2]
```diff
@@ -114,6 +116,42 @@ def normalize_vcs_url(repo_url, vcs_tool=None):
...
+    'github.com':    '/issues',
+    'gitlab.com':    '/-/issues',
+    'codeberg.org':  '/issues',
+    'bitbucket.org': '/issues',
+}
+
+
+def parse_vcs_urls(vcs_url):
+    '''
+    given a vcs_url, return (code_view_url, bug_tracking_url) when the
```
Comment (AyanSinhaMahapatra) [inline src/packagedcode/utils.py:130]: You misunderstand code view URLs, this should showthe source tree at a specific version/commit and not just the default code view.

Add this for pinned packages with exact versions only. No need to populate this for purls without versions.

File: tests/packagedcode/test_utils.py [hunk 3]
```diff
@@ -128,3 +129,45 @@ def test_normalize_vcs_url_does_not_fail_on_empty(self):
         assert normalize_vcs_url(None) == None
         assert normalize_vcs_url('') == None
```
Comment (AyanSinhaMahapatra) [inline tests/packagedcode/test_utils.py:130]: I'm surprises there are no test expectation changes, there should be lots of test files with updated URLs, can you look for/fix test failures?

**Review by AyanSinhaMahapatra — CHANGES_REQUESTED** [review]
Thanks @Kaushik-Kumar-CEG, see comments for your consideration.

File: src/packagedcode/models.py (hunk 1 above)
Comment (Kaushik-Kumar-CEG) [reply src/packagedcode/models.py:771]: Thanks for confirming!

File: src/packagedcode/utils.py (hunk 2 above)
Comment (Kaushik-Kumar-CEG) [reply src/packagedcode/utils.py]: good catch, fixed! code_view_url now points to the tagged source tree (e.g. /tree/1.2.3 for GitHub) and is only populated when the package has a concrete version

File: tests/packagedcode/test_utils.py (hunk 3 above)
Comment (Kaushik-Kumar-CEG) [reply tests/packagedcode/test_utils.py:130]: right, fixed. Updated the expected fixtures across the test suite (SCANCODE_REGEN_TEST_FIXTURES) — they now include the inferred URLs

**Comment by Kaushik-Kumar-CEG** [comment]
@AyanSinhaMahapatra 
Updated the PR based on your review. code_view_url now points to the version-specific source tree and is only set for packages with a concrete version. Test fixtures updated across the suite. All locals test passing

ready for review :)
